src/doc/en/developer/packaging.rst

```diff
@@ -1366,3 +1207,220 @@ must meet the following requirements:
+  for upstream.  Use them if possible, don't reinvent the wheel.
+
+- If the upstream project is maintained on GitHub, check if there is a Pull
+  Request that can be imported; see :ref:`section-spkg-patch-from-pr` below.
+
+- Sometimes it may seem as if you need to patch a (hand-written)
+  ``Makefile`` because it "hard-codes" some paths or compiler flags:
+
+  .. CODE-BLOCK:: diff
+
+      --- a/Makefile
+      +++ b/Makefile
+      @@ -77,7 +77,7 @@
+       # This is a Makefile.
+       # Handwritten.
+
+      -DESTDIR = /usr/local
+      +DESTDIR = $(SAGE_ROOT)/local
+       BINDIR   = $(DESTDIR)/bin
+       INCDIR   = $(DESTDIR)/include
+       LIBDIR   = $(DESTDIR)/lib
+
+  Don't use patching for that.  Makefile variables can be overridden
+  from the command-line.  Just use the following in ``spkg-install``:
+
+  .. CODE-BLOCK:: bash
+
+      $(MAKE) DESTDIR="$SAGE_ROOT/local"
+
+- If the upstream Makefile does not build shared libraries,
+  don't bother trying to patch it.
+
+  Autoconfiscate the package instead and use the standard facilities
+  of Automake and Libtool.  This ensures that the shared library build
+  is portable between Linux and macOS.
+
+- If you have to make changes to ``configure.ac`` or other source
+  files of the autotools build system (or if you are autoconfiscating
+  the package), then you can't use patching; make a :ref:`modified
+  tarball <section-spkg-src>` instead.
+
+- If the patch would be huge, don't use patching.  Make a
+  :ref:`modified tarball <section-spkg-src>` instead.
+
+- Otherwise, :ref:`maintain a set of patches
+  <section-spkg-patch-maintenance>`.
+
+
+.. _section-spkg-patch-from-pr:
+
+Patching sources by importing a pull request from GitHub
+--------------------------------------------------------
+
+Only ``normal`` packages can be patched; see :ref:`section-package-source-types`.
+If a Python package is currently a ``wheel`` package
+and you need to patch it, change it to a ``normal`` package first.
+
+In the easiest and quite commmon case, a pull request is already available on
+the upstream package's GitHub repository.
+
+For example, if https://github.com/discopt/cmr/pull/64 is the PR that we wish to use,
+change the URL to https://github.com/discopt/cmr/pull/64.patch and save this file
+in the ``patches/`` subdirectory of the package directory (create the subdirectory
+if it does not exist yet). Make sure that it has the ``.patch``
+file name extension; if your browser saved it with a ``.patch.txt`` extension,
+rename it.
+
+Modify the ``package-version.txt`` file to indicate the changed patch level; see
+:ref:`section-spkg-versioning`. This ensures that the package will be rebuilt,
+even though its upstream version did not change. This is important because when
+you open a pull request in the Sage repository that adds the patch, also
+our automatic test runs on GitHub Actions need to rebuild the packages,
+and likewise when other people are testing your pull request.
+
```

Comment:
This paragraph seems to belong to the next section.

Reply:
I've rewritten this a bit; not sure if this went in the direction that you hoped